incubator/haproxy-ingress/templates/controller-service.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: v1
+kind: Service
+metadata:
+{{- if .Values.controller.service.annotations }}
+  annotations:
+{{ toYaml .Values.controller.service.annotations | indent 4 }}
+{{- end }}
+  labels:
+{{- if .Values.controller.service.labels }}
+{{ toYaml .Values.controller.service.labels | indent 4 }}
+{{- end }}
+    run: {{ template "haproxy-ingress.name" . }}
+  name: {{ template "haproxy-ingress.controller.fullname" . }}
+  namespace: {{ .Values.namespace }}
+spec:
+  clusterIP: "{{ .Values.controller.service.clusterIP }}"
+  ports:
+    - name: http
+      port: 80
+      protocol: TCP
+      targetPort: {{ .Values.controller.service.targetPorts.http }}
+      {{- if (and (eq .Values.controller.service.type "NodePort") (not (empty .Values.controller.service.nodePorts.http))) }}
+      nodePort: {{ .Values.controller.service.nodePorts.http }}
+      {{- end }}
+    - name: https
+      port: 443
+      protocol: TCP
+      targetPort: {{ .Values.controller.service.targetPorts.https }}
+      {{- if (and (eq .Values.controller.service.type "NodePort") (not (empty .Values.controller.service.nodePorts.https))) }}
+      nodePort: {{ .Values.controller.service.nodePorts.https }}
+      {{- end }}
+  selector:
```

Comment:
Please refer [names-and-labels](https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md#names-and-labels) to set selector.

Reply:
Done